proxy-node-shared/src/main/java/uk/gov/ida/notification/saml/SamlObjectSigner.java

```diff
@@ -34,13 +38,8 @@ public SignatureSigningParameters getSigningParams() {
     }
 
     private  void logSigningRequest(String responseId, String signingProvider) {
-        try {
-            MDC.put("eidasResponseID", responseId);
-            MDC.put("signingProvider", signingProvider);
-            log.info("Signing eIDAS response");
-        } finally {
-            MDC.remove("eidasResponseID");
-            MDC.remove("signingProvider");
-        }
+        proxyNodeLogger.addContext(ProxyNodeMDCKey.EIDAS_REQUEST_ID, responseId);
```

Comment:
The proxyNodeLogger instance delegates to sl4j MDC static methods to add values to a ThreadLocal, so this would be threadsafe.

Not removing values from the MDC is intentional, the ProxyNodeLoggingFilter post-response `filter` method will clear down values we have set.

The `log` method delegates to a private final static logger within ProxyNodeLogger so this is threadsafe.

Reply:
> The proxyNodeLogger instance delegates to sl4j MDC static methods to add values to a ThreadLocal, so this would be threadsafe.
Sounds good 👍